common/src/main/java/com/wynntils/wc/custom/item/WynnItemStack.java

```diff
@@ -35,19 +36,37 @@ public List<Component> getOriginalTooltip() {
     }
 
     public void addProperty(ItemProperty property) {
-        if (getProperty(property.getClass()) != null) return; // don't allow duplicate properties of same type
+        if (getProperty(property.getClass()) != null) return; // don't allow duplicate properties
```

Comment:
I'm not sure I get this. How can we ever set multiple properties of the same class? Or what does `getProperties` do? 

(And, you now have a `hasProperty`, why not use it? ;-))

Reply:
the point is to prevent adding multiple instances of the same property, yes - probably not necessary, but might as well be safe. you're right, this should use `hasProperty`